BSP/Src/BSP_I2C.c

```diff
@@ -0,0 +1,180 @@
+
+#include "BSP_I2C.h"
+#include "BSP.h"
+#include "FreeRTOS.h"
+#include "queue.h"
+
+#define I2C_QUEUE_SIZE   64
+#define I2C_ITEM_LENGTH   sizeof(uint8_t)
+
+static StaticQueue_t __DataStaticQueue;
+static StaticQueue_t __MetaStaticQueue;
+static StaticQueue_t __ReceiveStaticQueue;
+
+
+uint8_t __DataBufferStorage[I2C_QUEUE_SIZE * I2C_ITEM_LENGTH];
+uint8_t __MetaBufferStorage[I2C_QUEUE_SIZE * I2C_ITEM_LENGTH * 2];
```

Comment:
thinking * 2 because the meta queue stores 1x address and 1x length for each non-register write?

Will need to implement more queues for the other types of functions (register/non register). Are we creating the queues on the BSP layer or is the caller making them? Evaluate.